apps/web/composables/useLocalization/__tests__/useLocalization.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { mockNuxtImport } from '@nuxt/test-utils/runtime';
+describe('useLocalization', () => {
+  const { useNuxtApp } = vi.hoisted(() => {
+    return {
+      useNuxtApp: vi.fn().mockImplementation(() => {
+        return {
+          $i18n: {
+            locale: ref('de'),
+            defaultLocale: 'en',
+            strategy: 'prefix_and_default',
+          }
+        }
+      })
+    }
+  });
+
+  mockNuxtImport('useNuxtApp', () => {
+    return useNuxtApp
+  });
+  
+
+  afterEach(() => {
+    // Restore mocks after each test
+    vi.clearAllMocks();
+  });
+
+  it('should get url without language prefix default strategy', async () => {
+    const { getCategoryUrlFromRoute } = useLocalization();
+    const res = getCategoryUrlFromRoute('/de/category/subcategory');
+    expect(res).toBe('/category/subcategory');
+  });
+
+  it('should test strategy "prefix"', () => {
```

Comment:
**issue**: I think this structure is misleading. The original `mockImplementation` is at the `describe` level, but here and in later tests, you change the setup at the `it` level. It would be better the `prefix_and_default` strategy to the first `it` statement.

Alternatively, you could use multiple `describe` statements, but this only makes sense if we have more than one test case per context.

Reply:
Moved the test up and removed one of the mockImplementations